docs/extendedusage.rst

```diff
@@ -344,6 +351,279 @@ would be ``/foo/bar/{id}``, and the absolute URI path would be
 Check out :doc:`api` for full definition of what is available for a ``resource`` object, and its associated attributes and objects.
 
 
+.. _nonramlparsing:
+
+Non-RAML Parsing
+----------------
+
+JSON
+^^^^
+
+For ``json`` filetypes, ``ramlfications`` will also parse ``$ref`` keywords and bring in the referenced objects according to both `Draft 3`_ and `Draft 4`_ JSON Schema definition.
+
+The following ``$ref`` values are supported:
+
+* :ref:`internalfrag`
+* :ref:`localfile`
+* :ref:`remotefile`
+
+.. _internalfrag:
+
+Internal fragments
+~~~~~~~~~~~~~~~~~~
+
+RAML File:
+
+.. code-block:: yaml
+
+    #%RAML 0.8
+    title: Sample API Demo - JSON Includes
+    version: v1
+    schemas:
+        - json: !include includes/ref_internal_fragment.json
+    baseUri: http://json.example.com
+    /foo:
+      displayName: foo resource
+
+``includes/ref_internal_fragment.json`` file:
+
+.. code-block:: json
+
+    {
+      "references": {
+        "internal": "yes"
+      },
+      "name": "foo",
+      "is_this_internal?": [{"$ref": "#/references/internal"}]
+    }
+
+
+``ramlfications`` would produce the following:
+
+.. code-block:: pycon
+
+    >>> RAML_FILE = "api.raml"
+    >>> api = parse(RAML_FILE)
+    >>> api.schemas
+    [{'json': {u'is_this_internal?': [u'yes'],
+    u'name': u'foo',
+    u'references': {u'internal': u'yes'}}}]
+
+.. _localfile:
+
+Local file with & without fragments
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Parsing references to local files via relative or absolute filepaths work fine, as well as prepending the URI with ``file:``, e.g. ``file:local.schema.json``.
+
+
+RAML File that includes a JSON file under ``schemas``:
+
+.. code-block:: yaml
+
+    #%RAML 0.8
+    title: Sample API Demo - JSON Includes
+    version: v1
+    schemas:
+        - jsonexample: !include local.schema.json
+    baseUri: http://json.example.com
+    /foo:
+      displayName: foo resource
+
+
+The included ``local.schema.json`` file that refers to another JSON file via a relative filepath and a fragment:
+
+.. code-block:: json
+
+    {
+      "$schema":"http://json-schema.org/draft-03/schema",
+      "type": "object",
+      "properties": {
+        "album_type": {
+          "type": "string",
+          "description": "The type of the album: one of 'album', 'single', or 'compilation'."
+        },
+        "artists": {
+            "type": "array",
+            "description": "The artists of the album. Each artist object includes a link in href to more detailed information about the artist.",
+            "items": [{ "$ref": "artist.schema.json#properties" }]
```

Comment:
side note - I just included a test that makes sure `file:` does work.  So the following cases should all work (all with or without `#fragments`):

* `{"$ref": "local.schema.json"}`
* `{"$ref": "file:local.schema.json"}`
* `{"$ref": "include/local.schema.json"}`
* `{"$ref": "file:include/local.schema.json"}`
* `{"$ref": "/absolute/path/include/local.schema.json"}`
* `{"$ref": "file:/absolute/path/include/local.schema.json"}`